homeassistant/components/konnected/config_flow.py

```diff
@@ -0,0 +1,732 @@
+            return self.async_abort(reason="unknown")
+
+        # swap out pin for zones in a io config
+        def pins_to_zones(config):
+            for zone in config:
+                if zone.get(CONF_PIN):
+                    zone[CONF_ZONE] = PIN_TO_ZONE[zone[CONF_PIN]]
+                    del zone[CONF_PIN]
+
+        if device_config.get(CONF_BINARY_SENSORS):
+            pins_to_zones(device_config[CONF_BINARY_SENSORS])
+
+        if device_config.get(CONF_SENSORS):
+            pins_to_zones(device_config[CONF_SENSORS])
+
+        if device_config.get(CONF_SWITCHES):
+            pins_to_zones(device_config[CONF_SWITCHES])
+
+        self.device_id = self.context["device_id"] = device_config["id"]
+        try:
+            self.host = device_config[CONF_HOST]
+            self.port = device_config[CONF_PORT]
+
+        except KeyError:
+            # cache config and wait for user input or discovery to provide host info
+            self.async_cache_config(device_config)
+            return await self.async_step_user()
+
+        # create the config entry
+        return await self.async_create_or_update_entry(device_config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input:
+            try:
+                self.host = user_input[CONF_HOST]
+                self.port = user_input[CONF_PORT]
+
+                # try to obtain the mac address from the device
+                status = await get_status(self.hass, self.host, self.port)
+                self.device_id = status.get("mac").replace(":", "")
+                self.model = status.get("name", KONN_MODEL)
+
+                # finish if we have a cached data set (partial config imported earlier)
+                if self.cached_config:
+                    config = copy.deepcopy(self.cached_config)
+                    try:
+                        config[CONF_HOST] = self.host
+                        config[CONF_PORT] = self.port
+                        return await self.async_create_or_update_entry(
+                            DEVICE_SCHEMA(config)
+                        )
+
+                    except vol.Invalid as err:
+                        _LOGGER.warning(
+                            "Invalid cached config..%s", humanize_error(config, err),
+                        )
+
+                return await self.async_step_io()
+
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+
+        return self.async_show_form(
+            step_id="user", data_schema=vol.Schema(DATA_SCHEMA_MANUAL), errors=errors
+        )
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered konnected panel.
+
+        This flow is triggered by the SSDP component. It will check if the
+        device is already configured and attempt to finish the config if not.
+        """
+        from homeassistant.components.ssdp import (
```

Comment:
All imports should be at the top of the module.

Reply:
👍